pkg/ims/repository/postgresql/users_repository.go

```diff
@@ -71,35 +71,59 @@ func (ur *UsersRepository) UpdateUser(u models.Users) error {
 
 func (ur *UsersRepository) DeleteUser(UserID int) error {
 
-	_, err := ur.db.Exec("DELETE FROM users WHERE id= $1", UserID)
+	_, err := ur.DB.Exec("DELETE FROM users WHERE id= $1", UserID)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (ur *UsersRepository) GetUserByEmail(email string) (*models.Users, error) {
+func (ur *UsersRepository) GetUserByEmail(email string) (*models.User, error) {
 
 	u := user{}
 
-	err := ur.db.Get(&u, "SELECT * FROM users WHERE email= $1", email)
+	err := ur.DB.Get(&u, "SELECT * FROM users WHERE email= $1", email)
 	if err != nil {
 		return nil, err
 	}
 
 	return ToUser(u)
 }
 
-func ToUser(privatUser user) (*models.Users, error) {
+// CheckEmail returns email if it exists in DB or empty string
+func (ur *UsersRepository) CheckEmail (email string) (string, error) {
+	var emailVal string
+
+	err := ur.DB.Get(&emailVal, "SELECT email FROM users WHERE email= $1", email)
+	if err != nil {
+		return emailVal, QueryError{queryErrorMessage, err}
+	}
+
+	return emailVal, nil
+}
+
+// GetEmailByToken returns email by token
+func (ur *UsersRepository) GetEmailByToken (token *models.EmailVerificationTokens) (string, error) {
+	var email string
+
+	err := ur.DB.Get(&email, "SELECT (SELECT email FROM users WHERE id = email_verification_tokens.id) FROM email_verification_tokens WHERE verification_token = $1", token.VerificationToken)
```

Comment:
please make an issue to simplify this query

Reply:
Issue created: https://github.com/omc-college/management-system/issues/101